pkg/controller/machine.go

```diff
@@ -339,6 +339,23 @@ func (c *controller) updateMachineState(machine *v1alpha1.Machine) (*v1alpha1.Ma
 func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driver) error {
 	glog.V(2).Infof("Creating machine %q, please wait!", machine.Name)
 
+	err := c.addBootstrapTokenToUserData(machine.Name, driver)
+	if err != nil {
+		glog.Errorf("Error while creating bootstrap token for machine %s: %s", machine.Name, err.Error())
+		lastOperation := v1alpha1.LastOperation{
+			Description:    "MCM message - " + err.Error(),
+			State:          v1alpha1.MachineStateFailed,
+			Type:           v1alpha1.MachineOperationCreate,
+			LastUpdateTime: metav1.Now(),
+		}
+		currentStatus := v1alpha1.CurrentStatus{
+			Phase:          v1alpha1.MachineFailed,
+			TimeoutActive:  false,
+			LastUpdateTime: metav1.Now(),
+		}
+		c.updateMachineStatus(machine, lastOperation, currentStatus)
+		return err
+	}
 	actualProviderID, nodeName, err := driver.Create()
 	if err != nil {
```

Comment:
Will you need to delete the bootstrap even when machine creation fails here? We could relay on the expiration time but deleting it explicitly would make it safer?

Reply:
You are right. The deletion shall occurs upon driver.Create() failure, too.